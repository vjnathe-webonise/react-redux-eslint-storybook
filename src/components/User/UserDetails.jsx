import { useParams } from 'react-router-dom';
import { Link } from 'react-router-dom';

export default function UserDetails() {
  const params = useParams();
  return (
    <>
      <Link to="/user">Back</Link>
      <h1>User Details of ID: {params.id} </h1>
    </>
  );
}
