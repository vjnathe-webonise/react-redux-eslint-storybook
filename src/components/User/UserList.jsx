import { Link } from 'react-router-dom';

export default function UserList() {
  return (
    <>
      <Link to="/">Home</Link>
      <h1>User List</h1>
      <ul>
        {[1, 2, 3].map((uid) => (
          <li key={uid}>
            <Link to={`/user/${uid}`}>User{uid}</Link>
          </li>
        ))}
      </ul>
    </>
  );
}
