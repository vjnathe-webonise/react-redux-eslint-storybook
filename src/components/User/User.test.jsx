import renderer from 'react-test-renderer';
import UserList from './UserList';
import UserDetails from './UserDetails';
import { BrowserRouter as Router } from 'react-router-dom';

it('UserList componenet renders correctly', () => {
  const tree = renderer
    .create(
      <Router>
        <UserList />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('UserDetails componenet renders correctly', () => {
  const tree = renderer
    .create(
      <Router>
        <UserDetails />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
