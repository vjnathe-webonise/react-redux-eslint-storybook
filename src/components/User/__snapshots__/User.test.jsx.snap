// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetails componenet renders correctly 1`] = `
Array [
  <a
    href="/user"
    onClick={[Function]}
  >
    Back
  </a>,
  <h1>
    User Details of ID: 
     
  </h1>,
]
`;

exports[`UserList componenet renders correctly 1`] = `
Array [
  <a
    href="/"
    onClick={[Function]}
  >
    Home
  </a>,
  <h1>
    User List
  </h1>,
  <ul>
    <li>
      <a
        href="/user/1"
        onClick={[Function]}
      >
        User
        1
      </a>
    </li>
    <li>
      <a
        href="/user/2"
        onClick={[Function]}
      >
        User
        2
      </a>
    </li>
    <li>
      <a
        href="/user/3"
        onClick={[Function]}
      >
        User
        3
      </a>
    </li>
  </ul>,
]
`;
